consensus/ibft/ibft.go

```diff
@@ -183,28 +190,105 @@ type ConsensusMechanism interface {
 	initializeHookMap()
 }
 
+type BaseConsensusMechanism struct {
+	// Reference to the main IBFT implementation
+	ibft *Ibft
+
+	// hookMap is the collection of registered hooks
+	hookMap map[HookType]func(interface{}) error
+
+	// Used for easy lookups
+	mechanismType MechanismType
+
+	// Available periods
+	From uint64
+	To   *uint64
+}
+
+// initializeParams initializes mechanism parameters from chain config
+func (base *BaseConsensusMechanism) initializeParams(params *IBFTFork) error {
+	if params == nil {
+		return nil
+	}
+
+	base.From = params.From.Value
+
+	if params.To != nil {
+		if params.To.Value < base.From {
+			return fmt.Errorf(`"to" must be grater than or equal to from: from=%d, to=%d`, base.From, params.To.Value)
```

Comment:
Nitpick warning:
Please break this up into multiple lines for readability, like with other `initializeParams` methods

Reply:
Fixed at 464cea9821e28327cb3c8e85cda47c21753feec5